src/pyramid/predicates.py

```diff
@@ -285,6 +285,14 @@ def __init__(self, val, config):
         else:
             self.val = set((val,))
 
+    __init__ = deprecated(
+        __init__,
+        'The new security policy has removed the concept of principals.  See '
+        'https://docs.pylonsproject.org/projects/pyramid/en/latest'
+        '/whatsnew-2.0.html#upgrading-authentication-authorization '
+        'for more information.',
+    )
+
```

Comment:
We should move this deprecation into `add_view` and `add_route` such that the `stacklevel` can be set appropriately to match user code. For reference, here is the `check_csrf` deprecation which was done in 1.10-branch. https://github.com/Pylons/pyramid/blob/bfa2ca0684c0821be8d50718169adc0a3dee8480/src/pyramid/config/views.py#L823-L834